Upstream: Accepted. https://git.reviewboard.kde.org/r/118670/

From 07924253350e8ed55f7e993bd3b0d30fb7627392 Mon Sep 17 00:00:00 2001
From: Bernd Steinhauser <berniyh@exherbo.org>
Date: Sun, 18 May 2014 07:17:49 +0200
Subject: [PATCH] Rename for coinstallability with kde4 version.

kfilemetadata already uses KF5 naming internally.
Start using the KF5 scheme externally as well to
avoid clashes with the kde4 version.
Install headers similar as other frameworks
---
 CMakeLists.txt                                     | 13 ++++---
 ...nfig.cmake.in => KF5FileMetaDataConfig.cmake.in |  2 +-
 autotests/CMakeLists.txt                           |  4 +-
 src/CMakeLists.txt                                 | 44 ++++++++++++----------
 src/extractors/CMakeLists.txt                      | 20 +++++-----
 5 files changed, 45 insertions(+), 38 deletions(-)
 rename KFileMetaDataConfig.cmake.in => KF5FileMetaDataConfig.cmake.in (91%)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5a9eefa..bb48ece 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,13 +12,14 @@ include(ECMAddTests)
 include(FeatureSummary)
 include(GenerateExportHeader)
 include(ECMSetupVersion)
+include(ECMGenerateHeaders)
 
 set(KF5_VERSION "4.97.0")
 set(REQUIRED_QT_VERSION "5.2.0")
 
 ecm_setup_version(${KF5_VERSION}
                   VARIABLE_PREFIX KFILEMETADATA
-                  PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/KFileMetaDataConfigVersion.cmake"
+                  PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/KF5FileMetaDataConfigVersion.cmake"
                   SOVERSION 5)
 
 # Build dependencies
@@ -67,19 +68,19 @@ if (BUILD_TESTING)
 endif()
 
 # Config files
-set(CMAKECONFIG_INSTALL_DIR "${CMAKECONFIG_INSTALL_PREFIX}/KFileMetaData")
+set(CMAKECONFIG_INSTALL_DIR "${CMAKECONFIG_INSTALL_PREFIX}/KF5FileMetaData")
 
 include(ECMPackageConfigHelpers)
 
 ecm_configure_package_config_file(
-  "${CMAKE_CURRENT_SOURCE_DIR}/KFileMetaDataConfig.cmake.in"
-  "${CMAKE_CURRENT_BINARY_DIR}/KFileMetaDataConfig.cmake"
+  "${CMAKE_CURRENT_SOURCE_DIR}/KF5FileMetaDataConfig.cmake.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/KF5FileMetaDataConfig.cmake"
   INSTALL_DESTINATION  ${CMAKECONFIG_INSTALL_DIR}
 )
 
 install(FILES
-  "${CMAKE_CURRENT_BINARY_DIR}/KFileMetaDataConfig.cmake"
-  "${CMAKE_CURRENT_BINARY_DIR}/KFileMetaDataConfigVersion.cmake"
+  "${CMAKE_CURRENT_BINARY_DIR}/KF5FileMetaDataConfig.cmake"
+  "${CMAKE_CURRENT_BINARY_DIR}/KF5FileMetaDataConfigVersion.cmake"
   DESTINATION "${CMAKECONFIG_INSTALL_DIR}"
   COMPONENT devel
 )
diff --git a/KFileMetaDataConfig.cmake.in b/KF5FileMetaDataConfig.cmake.in
similarity index 91%
rename from KFileMetaDataConfig.cmake.in
rename to KF5FileMetaDataConfig.cmake.in
index b4d1c93..551441c 100644
--- a/KFileMetaDataConfig.cmake.in
+++ b/KF5FileMetaDataConfig.cmake.in
@@ -11,7 +11,7 @@ get_filename_component(rootDir ${_currentDir}/@relInstallDir@ ABSOLUTE) # The in
 set(KFILEMETADATA_INSTALL_PREFIX "${rootDir}")
 set(KFILEMETADATA_INCLUDE_DIR "@INCLUDE_INSTALL_DIR@")
 
-include(${_currentDir}/KFileMetaDataTargetsWithPrefix.cmake)
+include(${_currentDir}/KF5FileMetaDataTargetsWithPrefix.cmake)
 
 # Set the library variable
 set(KFILEMETADATA_LIBRARY KF5::FileMetaData)
diff --git a/autotests/CMakeLists.txt b/autotests/CMakeLists.txt
index c657a70..bc6cd19 100644
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -10,7 +10,7 @@ set(indexerextractor_SRCS
 )
 ecm_add_test(${indexerextractor_SRCS}
     TEST_NAME "indexextractortest"
-    LINK_LIBRARIES Qt5::Test KF5::Service kfilemetadata
+    LINK_LIBRARIES Qt5::Test KF5::FileMetaData KF5::Service
 )
 
 #
@@ -18,6 +18,6 @@ ecm_add_test(${indexerextractor_SRCS}
 #
 ecm_add_test(propertyinfotest.cpp
     TEST_NAME "propertyinfotest"
-    LINK_LIBRARIES Qt5::Test kfilemetadata
+    LINK_LIBRARIES Qt5::Test KF5::FileMetaData
 )
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 82dbd5c..9bf32e6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,4 +1,4 @@
-add_library(kfilemetadata SHARED
+add_library(KF5FileMetaData SHARED
     extractionresult.cpp
     extractorplugin.cpp
     extractorpluginmanager.cpp
@@ -6,9 +6,9 @@ add_library(kfilemetadata SHARED
     typeinfo.cpp
 )
 
-add_library(KF5::FileMetaData ALIAS kfilemetadata)
+add_library(KF5::FileMetaData ALIAS KF5FileMetaData)
 
-target_link_libraries(kfilemetadata
+target_link_libraries(KF5FileMetaData
   PUBLIC
     Qt5::Core
   PRIVATE
@@ -16,35 +16,41 @@ target_link_libraries(kfilemetadata
     KF5::I18n
 )
 
-generate_export_header(kfilemetadata BASE_NAME KFileMetaData EXPORT_FILE_NAME kfilemetadata_export.h)
-set_target_properties(kfilemetadata PROPERTIES
+generate_export_header(KF5FileMetaData BASE_NAME KFileMetaData EXPORT_FILE_NAME kfilemetadata_export.h)
+set_target_properties(KF5FileMetaData PROPERTIES
     VERSION ${KFILEMETADATA_VERSION_STRING}
     SOVERSION ${KFILEMETADATA_SOVERSION}
     EXPORT_NAME FileMetaData
 )
-target_include_directories(kfilemetadata INTERFACE "$<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}>")
+target_include_directories(KF5FileMetaData INTERFACE "$<INSTALL_INTERFACE:${KF5_INCLUDE_INSTALL_DIR}/KFileMetaData>")
+
+ecm_generate_headers(KF5FileMetaData_HEADERS
+    HEADER_NAMES
+    ExtractionResult
+    ExtractorPlugin
+    ExtractorPluginManager
+    Properties
+    PropertyInfo
+    Types
+    TypeInfo
+
+    REQUIRED_HEADERS KF5FileMetaData_HEADERS
+)
 
-install(TARGETS kfilemetadata ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS KF5FileMetaData ${INSTALL_TARGETS_DEFAULT_ARGS})
 
-install(TARGETS kfilemetadata EXPORT KFileMetaDataTargets ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS KF5FileMetaData EXPORT KFileMetaDataTargets ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(FILES kfilemetadataextractor.desktop DESTINATION ${SERVICETYPES_INSTALL_DIR})
 
 install(EXPORT KFileMetaDataTargets
         NAMESPACE KF5::
-        DESTINATION ${LIB_INSTALL_DIR}/cmake/KFileMetaData
-        FILE KFileMetaDataTargetsWithPrefix.cmake)
+        DESTINATION ${LIB_INSTALL_DIR}/cmake/KF5FileMetaData
+        FILE KF5FileMetaDataTargetsWithPrefix.cmake)
 
 install(FILES
-    extractionresult.h
-    extractorplugin.h
-    extractorpluginmanager.h
-    properties.h
-    propertyinfo.h
-    types.h
-    typeinfo.h
     ${CMAKE_CURRENT_BINARY_DIR}/kfilemetadata_export.h
-
-    DESTINATION ${INCLUDE_INSTALL_DIR}/kfilemetadata COMPONENT Devel
+    ${KF5FileMetaData_HEADERS}
+    DESTINATION ${KF5_INCLUDE_INSTALL_DIR}/KFileMetaData COMPONENT Devel
 )
 
 add_subdirectory(extractors)
diff --git a/src/extractors/CMakeLists.txt b/src/extractors/CMakeLists.txt
index 0099c08..7dc7dd4 100644
--- a/src/extractors/CMakeLists.txt
+++ b/src/extractors/CMakeLists.txt
@@ -4,7 +4,7 @@ if(POPPLER_QT5_FOUND)
     add_library(kfilemetadata_popplerextractor MODULE popplerextractor.cpp)
 
     target_link_libraries(kfilemetadata_popplerextractor
-        kfilemetadata
+        KF5::FileMetaData
         KF5::Service
         ${POPPLER_QT5_LIBRARIES}
     )
@@ -25,7 +25,7 @@ if(TAGLIB_FOUND)
     add_library(kfilemetadata_taglibextractor MODULE taglibextractor.cpp )
 
     target_link_libraries( kfilemetadata_taglibextractor
-        kfilemetadata
+        KF5::FileMetaData
         KF5::Service
         ${TAGLIB_LIBRARIES}
     )
@@ -47,7 +47,7 @@ if(EXIV2_FOUND)
     add_library(kfilemetadata_exiv2extractor MODULE exiv2extractor.cpp)
 
     target_link_libraries(kfilemetadata_exiv2extractor
-        kfilemetadata
+        KF5::FileMetaData
         KF5::Service
         ${EXIV2_LIBRARIES}
     )
@@ -68,7 +68,7 @@ if(FFMPEG_FOUND)
     add_library(kfilemetadata_ffmpegextractor MODULE ffmpegextractor.cpp)
 
     target_link_libraries(kfilemetadata_ffmpegextractor
-        kfilemetadata
+        KF5::FileMetaData
         KF5::Service
         ${FFMPEG_LIBRARIES}
     )
@@ -90,7 +90,7 @@ if(EPUB_FOUND)
     add_library(kfilemetadata_epubextractor MODULE epubextractor.cpp)
 
     target_link_libraries(kfilemetadata_epubextractor
-        kfilemetadata
+        KF5::FileMetaData
         KF5::Service
         ${EPUB_LIBRARIES}
     )
@@ -111,7 +111,7 @@ endif(EPUB_FOUND)
 add_library(kfilemetadata_plaintextextractor MODULE plaintextextractor.cpp )
 
 target_link_libraries( kfilemetadata_plaintextextractor
-    kfilemetadata
+    KF5::FileMetaData
     KF5::Service
 )
 
@@ -130,7 +130,7 @@ DESTINATION ${PLUGIN_INSTALL_DIR})
 add_library(kfilemetadata_odfextractor MODULE odfextractor.cpp)
 
 target_link_libraries(kfilemetadata_odfextractor
-    kfilemetadata
+    KF5::FileMetaData
     Qt5::Core
     Qt5::Xml
     KF5::Service
@@ -152,7 +152,7 @@ DESTINATION ${PLUGIN_INSTALL_DIR})
 add_library(kfilemetadata_office2007extractor MODULE office2007extractor.cpp)
 
 target_link_libraries(kfilemetadata_office2007extractor
-    kfilemetadata
+    KF5::FileMetaData
     Qt5::Core
     Qt5::Xml
     KF5::Service
@@ -174,7 +174,7 @@ DESTINATION ${PLUGIN_INSTALL_DIR})
 add_library(kfilemetadata_officeextractor MODULE officeextractor.cpp)
 
 target_link_libraries(kfilemetadata_officeextractor
-    kfilemetadata
+    KF5::FileMetaData
     KF5::Service
 )
 
@@ -194,7 +194,7 @@ if (QMOBIPOCKET_FOUND)
 
     include_directories(${QMOBIPOCKET_INCLUDE_DIR})
     target_link_libraries(kfilemetadata_mobiextractor
-        kfilemetadata
+        KF5::FileMetaData
         KF5::Service
         ${QMOBIPOCKET_LIBRARIES}
     )
-- 
2.0.0

